libs/common/src/services/indexer/indexer.service.ts

```diff
@@ -96,4 +96,17 @@ export class IndexerService {
 
     return logs;
   }
+
+  @LogPerformanceAsync(MetricsEvents.SetIndexerDuration)
+  public async getTxDetails(txHashes: string[]) {
+    const query = ElasticQuery.create()
+      .withPagination({ from: 0, size: 10000 })
+      .withMustCondition([
+        QueryType.Should(txHashes.map(txHash => new MatchQuery("_id", txHash))),
```

Comment:
lets split it in batch of 100 txs i know there is a limitation of how many queries a mustCondition can have